drivers/timer/stm32_lptim_timer.c

```diff
@@ -79,6 +97,26 @@ static void lptim_irq_handler(const struct device *unused)
 	}
 }
 
+static void lptim_set_autoreload(uint32_t arr)
+{
+	uint32_t autoreload;
+
+	/* Update autoreload register */
+	autoreload_next = arr;
+	if (!autoreload_ready)
+		return;
+
+	autoreload = LL_LPTIM_GetAutoReload(LPTIM1);
+
+	/* The ARR register ready, we could set it directly */
+	if ((autoreload_next > 0) && (autoreload_next != autoreload)) {
```

Comment:
Maybe, this could still be optimized  :
```
suggestion
```
> if ((autoreload_next > 0) && (autoreload_next != autoreload)) {

> if ((arr > 0) && (arr != LL_LPTIM_GetAutoReload(LPTIM1))) {



```